src/components/Login/store/reducer.js

```diff
@@ -0,0 +1,26 @@
+import {
+  USERNAME_INPUT_VALUE,
+  PASSWORD_INPUT_VALUE,
+  CLEAR_PASSWORD_VALUE,
+  FORM_SUBMIT
+} from './types';
+
+const initialState = {
+  password: '',
+  username: '',
+};
+
+export default function loginReducer(state = initialState, action) {
+  switch (action.type) {
+    case USERNAME_INPUT_VALUE:
+      return { ...state, username: action.payload };
+    case PASSWORD_INPUT_VALUE:
+      return { ...state, password: action.payload };
+    case CLEAR_PASSWORD_VALUE:
+      return { ...state, password: '' };
+    case FORM_SUBMIT:
+      return { ...state };
+    default:
+      return state;
+  }
+}
```

Comment:
Is it good to have store inside Login component?
I think it will be better to have extra folder store in src dir and login folder inside it